File: Lecture8-Erosion/main.cpp
#include <opencv4/opencv2/opencv.hpp>

using namespace cv;

void erosion()
{
    Mat img = imread("/home/berkan/CLionProjects/OpenCV/Ders8_Erozyon/word.jpg", IMREAD_GRAYSCALE);
    namedWindow("Erosion", WINDOW_AUTOSIZE);
    threshold(img, img, 128, 255, THRESH_BINARY);

    int min = 0;
    createTrackbar("Erosion", "Erosion", &min, 5);

    Mat filter, erosionImg;
    filter = Mat::ones(Size(3,3), img.type());

    while (true)
    {
        erode(img, erosionImg, filter, Point(-1, -1), min, 0, Scalar(255, 255, 255));
        imshow("Erosion", erosionImg);
        if (waitKey(10) == 27)
        {
            break;
        }
    }
    destroyAllWindows();
}

int main() {
    erosion();
    return 0;
}

File: Lecture8-Erosion/main.py
import cv2
import numpy as np

def nothing():
    pass

def erozyon():
    img = cv2.imread("cumle.jpg", cv2.IMREAD_GRAYSCALE)
    cv2.namedWindow("Erozyon", cv2.WINDOW_AUTOSIZE)
    thres = cv2.threshold(img, 128 ,255, cv2.THRESH_BINARY)

    filtre = np.ones((3,3)) #3 satır 3 sütundan oluşan 1ler matrisi

    cv2.createTrackbar("Asindir", "Erozyon", 0, 15, nothing)

    while True:
        asindirmaDegeri = cv2.getTrackbarPos("Asindir", "Erozyon")
        erodeImg = cv2.erode(img, filtre, iterations=asindirmaDegeri)
        cv2.imshow("Erozyon", erodeImg)

        if cv2.waitKey(10) == 27:
            break

    cv2.destroyAllWindows()

if __name__ == '__main__':
    erozyon()